Modules/CitiesViewController/CitiesViewController.swift

```diff
@@ -0,0 +1,248 @@
+//
+//  ViewController.swift
+//  WeatherDemoApp
+//
+//  Created by ahmed on 01/04/2024.
+//
+
+import UIKit
+import Combine
+
+final class CitiesViewController: UITableViewController {
+    
+    private let viewModel: CitiesViewModel!
+    private var gradientView: CAGradientLayer?
+    private var disposables = Set<AnyCancellable>()
+    
+    init(viewModel: CitiesViewModel) {
+        self.viewModel = viewModel
+        super.init(style: .insetGrouped)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    // MARK: Life Cycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationController?.setNavigationBarHidden(true, animated: false)
+        
+        setupRightBtn()
+        setupPageTitleLbl()
+        addGradient()
+        
+        addPageTitleLabelConstaints()
+        addBtnViewConstaints()
+        addBottomImageConstaints()
+        
+        setupTableView()
+        
+        assignViewModelClosures()
+        
+        /// watch RealmDB on Simulator
+        /// print(FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!.path())
+    }
+    
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        // to activate gradient
+        gradientView?.frame = tableView.bounds
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        tableView.bringSubviewToFront(rightBtn)
+    }
+    
+    // MARK: View Configurators
+    private func setupTableView() {
+        tableView.rowHeight = 60
+        tableView.showsVerticalScrollIndicator = false
+        tableView.backgroundColor = Colors.pageGradientFirstColor
+        tableView.register(
+            CityTableCell.self,
+            forCellReuseIdentifier: CityTableCell.identifier
+        )
+        //        tableView.bounces = false
+    }
+    private func addGradient() {
+        let pageGradient = Colors.pageGradient
+        
+        gradientView = tableView.setGradient(colors: pageGradient.colors,
+                                             locations: pageGradient.locations,
+                                             startPoint: pageGradient.startPoint,
+                                             endPoint: pageGradient.endPoint)
+    }
+    
+    private func setupPageTitleLbl() {
+        pageTitleLbl.setTitle("Cities")
+    }
+    
+    // MARK: Navigation Methods
+    private func presentWeatherDetailScreen(_ cityWeatherInfo: LocalStorageCity) {
+        let detailVC = CityWeatherEntriesVC(cityWeatherInfo: cityWeatherInfo)
+        navigationController?.pushViewController(detailVC, animated: true)
+    }
+    
+    private func setupRightBtn() {
+        /// Action
+        let addBtnAction: VoidCallback = { [unowned self] in
```

Comment:
```suggestion
        let addBtnAction: VoidCallback = { [weak self] in
```
This can crash the app. Better safe than sorry.